crates/aranya-client/tests/tests.rs

```diff
@@ -132,19 +133,22 @@ fn trim(mut d: u128, mut width: usize) -> (u128, usize) {
     (d, width)
 }
 
-/// Repeatedly calls `poll_afc_data`, followed by `handle_afc_data`, until all
-/// of the clients are pending.
+/// Repeatedly calls `poll_data`, followed by `handle_data`, until all of the
+/// clients are pending.
+// TODO: alternative to select!{} to resolve lifetime issues
 macro_rules! do_poll {
     ($($client:expr),*) => {
         debug!(
             clients = stringify!($($client),*),
             "start `do_poll`",
         );
         loop {
+            let mut afcs = [ $($client.afc()),* ];
+            let mut afcs = afcs.iter_mut();
             tokio::select! {
                 biased;
-                $(data = $client.poll_afc_data() => {
-                    $client.handle_afc_data(data?).await?
+                $(data = afcs.next().unwrap().poll_data() => {
```

Comment:
Please add a comment about why you're using `unwrap` and why it doesn't panic. Or, better, use `next().assume("...")?` from `buggy::BugExt`.